meta-mender-qemu/scripts/docker/extract_fs

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+failure() {
+    echo "Failed! Make sure you are using a privileged container."
+    exit 1
+}
+
+DEV=$(losetup -f)
+IMG=$(ls /*img | head -n 1)
+OUTPUT=/output/$(sed -e 's/\.[^.]*img$/.ext4/' <<<$(basename $IMG))
+
+losetup $DEV $IMG || failure
+kpartx -a $DEV || failure
+echo "Copying filesystem image to $OUTPUT"
+dd if=/dev/mapper/$(basename $DEV)p2 of=$OUTPUT bs=8M status=none
+chown $(stat -c '%u:%g' /output) $OUTPUT
+
+kpartx -d $DEV
+losetup -d $DEV
```

Comment:
Originally I tried to avoid `set -e` to make sure that the cleanup steps were always carried out, but you're right, I think it's better to be safe and add it. I'll add the cleanup steps to a `trap` instead.

Reply:
That would be cleaner, I think